Source/JavaScriptCore/wasm/js/JSWebAssemblyArray.h

```diff
@@ -86,7 +92,23 @@ class JSWebAssemblyArray final : public JSNonFinalObject {
 
     void set(VM& vm, uint32_t index, EncodedJSValue value)
     {
-        switch (m_elementType.kind) {
+        if (m_elementType.type.is<Wasm::PackedType>()) {
+            // `value` is assumed to be an unboxed int64; truncate it to either 8 or 16 bits
```

Comment:
can we assert that this is valid?

Reply:
Added a similar assertion as above, `ASSERT(value <= UINT32_MAX);`